src/main/java/com/odysseusinc/arachne/portal/util/SubmissionHelper.java

```diff
@@ -441,14 +412,9 @@ public void updateExtendInfo(Submission submission) {
             submission.setResultInfo(result);
         }
 
-        private ArachneFileMeta findAnalysisPackage(Submission submission) {
-
-            final String analysisPackMask = String.format("%s-%%.zip", CommonAnalysisType.INCIDENCE.getCode());
-            List<ArachneFileMeta> packageFiles = searchFiles(submission, analysisPackMask);
-            if (packageFiles.isEmpty()) {
-                packageFiles = searchFiles(submission, "IncidenceRate%.zip");
-            }
-            return packageFiles.isEmpty() ? null : packageFiles.get(0);
+        private ArachneFileMeta findAnalysisSpec(Submission submission) {
+            List<ArachneFileMeta> files = searchFiles(submission, STUDY_SPECIFICATION_JSON);
+            return files.isEmpty() ? null : files.get(0);
```

Comment:
`return files.stream().findFirst().orElse(null)`
Or even better, return optional and use map() / ifPresent()

Reply:
done